modules/rostests/apitests/crt/__dtoi64.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * PROJECT:         ReactOS API tests
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:         Tests for __dtoi64
+ * COPYRIGHT:       Copyright 2021 Roman Masanin <36927roma@gmail.com>
+ */
+
+#include <apitest.h>
+
+typedef struct _DTOI64_TEST_DATA
+{
+    union
+    {
+        double double_value;
+        unsigned long long double_raw;
+    } test_data;
+    long long expected_longlong;
+} DTOI64_TEST_DATA;
+
+START_TEST(__dtoi64)
+{
+    DTOI64_TEST_DATA test_array[] = 
+    {
+        { 1383034209.0, 1383034209LL }, /* test 32bit number */
+        { -1383034209.0, -1383034209LL }, /* test negative 32bit number */
+        { 1383034209.1383034209, 1383034209LL }, /* test rounding 32bit */
+        { -1383034209.1383034209, -1383034209LL }, /* test negative rounding 32bit */
+        { 1383034209.83034209, 1383034209LL }, /* test rounding up 32bit */
+        { -1383034209.83034209, -1383034209LL }, /* test negative rounding up 32bit */
+        { 354056757614.0, 354056757614LL }, /* test 64bit int */
+        { -354056757614.0, -354056757614LL }, /* test negative 64bit int */
+        { 354056757614.83034209, 354056757614LL }, /* test 64bit rounding */
+        { 18445937028656326656.0, 0x8000000000000000LL }, /* test unsigned 64bit */
+        { 1.0000001, 1LL },
+        { 0.0000001, 0LL },
+        { -0.0000001, 0LL },
+        
+        /* special values tests */
+        
+        { -0.0, 0LL }, /* test -0 */
```

Comment:
I wonder, whether compiler will actually treat -0.0 as negative, or whether it might implicitly convert it to 0. Have you ever tested that?

Reply:
for double both +0 and -0 exists))